app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,70 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign_to = nil, cat = nil, objects = nil)
+    @assign_to = assign_to
+    @cat = cat
+    @objects = objects
+    build = false unless @assign_to
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    identifier = (object.name.presence || object.description)
+    if object.kind_of?(MiddlewareServer)
+      identifier += "-" + object.hostname
+    end
+
+    if @assign_to.ends_with?("-tags")
```

Comment:
How do you mean it? I thought that x_get_tree_roots is only for retrieving data, how you want to set nodes here, nodes are not existing here yet? Do you want only to get rid of the evaluation of the condition from two places to one?